### PR TITLE
Bug fixes and improvements

### PR DESCRIPTION
**Pull request by mbertrand**
Move gaia.core constants and functions to gaia.__init__ : by making this change, referring to the module configuration as 'gaia.config' seems cleaner than 'gaia.core.config' for example.

gaia.core.GaiaException => gaia.GaiaException
gaia.core.get_plugins() => gaia.get_plugins()

etc.

Also:
- Standardized transformation of vector data (reprojection and JSON output) and places in a VectorMixin class.

- Improved function used to get/guess data projections.

- Added more unit tests.

### CODE REVIEW

**Comment by aashish24** [comment]
@mbertrand is this ready for review?

**Comment by mbertrand** [comment]
@aashish24 yes, thanks

File: gaia/geo/gdal_functions.py [hunk 1]
```diff
@@ -475,35 +476,38 @@ def gen_zonalstats(zones_json, raster):
     targetSR = osr.SpatialReference()
     targetSR.ImportFromWkt(raster.GetProjectionRef())
     coordTrans = osr.CoordinateTransformation(sourceSR, targetSR)
-
+    # TODO: Use a multiprocessing pool to process features more quickly
```
Comment (aashish24) [inline gaia/geo/gdal_functions.py]: perhaps we can use dask?

**Comment by aashish24** [comment]
@mbertrand mostly looks reasonable to me LGTM

File: gaia/geo/gdal_functions.py (hunk 1 above)
Comment (mbertrand) [inline gaia/geo/gdal_functions.py]: @aashish24, @fx2323 from our DC office is looking into that.

**Pull request merged**